### ISSUE TITLE
Creation of dynamic property Psalm\Type\MutableUnion::$propagate_parent_nodes is deprecated

### ISSUE THREAD

**Issue by buismaarten** [vimeo/psalm#8767]
PHP version: 8.2

Output:
```
Target PHP version: 8.2 (inferred from current PHP version) Extensions enabled:  (unsupported extensions: )
Scanning files...
Uncaught RuntimeException: PHP Error: Creation of dynamic property Psalm\Type\MutableUnion::$propagate_parent_nodes is deprecated in /var/www/vendor/vimeo/psalm/src/Psalm/Type/UnionTrait.php:70 for command with CLI args "./vendor/bin/psalm" in /var/www/vendor/vimeo/psalm/src/Psalm/Internal/ErrorHandler.php:77
Stack trace:
#0 /var/www/vendor/vimeo/psalm/src/Psalm/Type/UnionTrait.php(70): Psalm\Internal\ErrorHandler::Psalm\Internal\{closure}()
#1 /var/www/vendor/vimeo/psalm/src/Psalm/Type/Union.php(340): Psalm\Type\MutableUnion->__construct()
#2 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeNodeScanner.php(851): Psalm\Type\Union->getBuilder()
#3 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/Reflector/FunctionLikeNodeScanner.php(215): Psalm\Internal\PhpVisitor\Reflector\FunctionLikeNodeScanner->getTranslatedFunctionParam()
#4 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/PhpVisitor/ReflectorVisitor.php(236): Psalm\Internal\PhpVisitor\Reflector\FunctionLikeNodeScanner->start()
#5 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(123): Psalm\Internal\PhpVisitor\ReflectorVisitor->enterNode()
#6 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#7 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#8 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(146): PhpParser\NodeTraverser->traverseNode()
#9 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#10 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#11 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#12 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(114): PhpParser\NodeTraverser->traverseArray()
#13 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(223): PhpParser\NodeTraverser->traverseNode()
#14 /var/www/vendor/nikic/php-parser/lib/PhpParser/NodeTraverser.php(91): PhpParser\NodeTraverser->traverseArray()
#15 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Scanner/FileScanner.php(88): PhpParser\NodeTraverser->traverse()
#16 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(578): Psalm\Internal\Scanner\FileScanner->scan()
#17 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(807): Psalm\Internal\Codebase\Scanner->scanFile()
#18 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(452): Psalm\Internal\Codebase\Scanner->scanAPath()
#19 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Codebase/Scanner.php(305): Psalm\Internal\Codebase\Scanner->scanFilePaths()
#20 /var/www/vendor/vimeo/psalm/src/Psalm/Config.php(2327): Psalm\Internal\Codebase\Scanner->scanFiles()
#21 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(444): Psalm\Config->visitComposerAutoloadFiles()
#22 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Analyzer/ProjectAnalyzer.php(654): Psalm\Internal\Analyzer\ProjectAnalyzer->visitAutoloadFiles()
#23 /var/www/vendor/vimeo/psalm/src/Psalm/Internal/Cli/Psalm.php(375): Psalm\Internal\Analyzer\ProjectAnalyzer->check()
#24 /var/www/vendor/vimeo/psalm/psalm(9): Psalm\Internal\Cli\Psalm::run()
#25 /var/www/vendor/bin/psalm(120): include('...')
#26 {main}
(Psalm 5.0.0-rc1@8f39de9001c995eb203cee3399307570f322076a crashed due to an uncaught Throwable)
```

**Comment by buismaarten** [comment]
It seems that the problem occurred from switching from PHP 8.1 to 8.2 without reinstalling the Composer packages.

**Comment by weirdan** [comment]
This was fixed in #8755
